.travis.yml

```diff
@@ -1,22 +1,43 @@
-sudo: false
-language: scala
-addons:
-  hostname: localhost
-jdk:
-  - oraclejdk8
-scala:
-  - 2.11.12
-  - 2.12.8
-before_install:
-  - docker pull daunnc/openjdk-gdal:2.4.0
+# Required to clean up sbt caches written by container image as root
+sudo: required
+
+# https://github.com/travis-ci/travis-ci/issues/2856
+language: bash
+
 services:
   - docker
+
+before_install:
+  - docker pull quay.io/azavea/openjdk-gdal:2.3.2-jdk8-slim 
+
+env:
+  - SCALA_VERSION=2.11.12
+  - SCALA_VERSION=2.12.8
+
+# These directories are cached to S3 at the end of the build
 cache:
   directories:
-  - $HOME/.ivy2
-  - $HOME/.sbt
+    - $HOME/.ivy2/cache
+    - $HOME/.sbt/boot/
+    - $HOME/.coursier/cache/v1
+
 script:
-  - .travis/build-and-test-docker.sh
+  - scripts/test
+
+before_cache:
+  # Tricks to avoid unnecessary cache updates
+  - find $HOME/.sbt -name "*.lock" | sudo xargs rm
+  - find $HOME/.ivy2 -name "ivydata-*.properties" | sudo xargs rm
+  - sudo rm -f $HOME/.ivy2/.sbt.ivy.lock
+
+deploy:
+  provider: script
+  script: bash scripts/cipublish
```

Comment:
It can't use just `./scripts/cipublish`?

Reply:
I'm not sure, I was following their documentation which only references invoking scripts with `$ bash ...`: 

https://docs.travis-ci.com/user/deployment/script/

Reply:
@rbreslow I think it uses the default shell if you won't specify `bash`